### PR TITLE
Add support for multiple hash types

### LINKED ISSUE #3: Support hashes besides MD5
# 🚀 Feature Proposal

Add the ability to scan for other hash types besides MD5, such as SHA-1 and SHA-256.

## Motivation

When we are asked to scan for Indicators of Compromise (IOCs), we occasionally are given SHA-1 and SHA-256 hashes, in addition to MD5 hashes.

## Example

Sample IOC hashes:
```
SHA256: b509f8545501588ecd828f970d91afc7c4aa6e238e838bd6a08ee2cd920fbe98
SHA-1:  31B54AEBDAF5FBC73A66AC41CCB35943CC9B7F72
SHA-1:  50973A3FC57D70C7911F7A952356188B9939E56B
SHA-1:  244EB62B9AC30934098CA4204447440D6FC4E259
SHA-1:  5C8F83CC4FF57E7C67925DF4D9DAABE5D0CC07E2
```

## Pitch

It will give us more comprehensive scanning capabilities.

### PR DESCRIPTION
**Pull request by mcdonnnj**
This PR does two things:
1.  Switches to a pure Python implementation of traversing the filesystem and generating hashes which gives better cross-platform support (looking at you Windows).
2.  Adds support for SHA1 and SHA256 hashes and makes it straightforward to add additional hash support in the future as long as it is supported by Python's hashlib module. This resolves #3

### CODE REVIEW

File: src/ioc_scan/ioc_scanner.py [hunk 1]
```diff
@@ -44,11 +43,29 @@
...
+    """Generate MD5, SHA1, and SHA256 hashes for a given file."""
+    hash_md5 = hashlib.md5()  # nosec
+    hash_sha1 = hashlib.sha1()  # nosec
+    hash_sha256 = hashlib.sha256()
+
+    # try except to eat filesystem errors like Permission Denied etc
+    try:
+        with open(file, "rb") as f:
+            # read it in chunks so memory use isn't outlandish
+            for chunk in iter(lambda: f.read(4096), b""):
```
Comment (jsf9k) [inline src/ioc_scan/ioc_scanner.py:67]: Sweet use of `lambda`!

**Review by jsf9k — APPROVED** [review]
These are tremendous improvements, and the script is much more extensible now.  Nice work!

**Review by dav3r — APPROVED** [review]
Very nice job on this @mcdonnnj #️⃣ #️⃣ #️⃣ 👍

**Review by felddy — APPROVED** [review]
Stellar!  ⭐️ 🌟

**Pull request merged**